### PR TITLE
experiment to check if SyncWAL works on Windows

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Enable RocksDB SyncWAL() calls on Windows as well.
Previously they were disabled for Windows but enabled for all other platforms. For Windows, there was a workaround. The reason for not calling SyncWAL() on Windows previously was that RocksDB reported the operation as unsupported.

Tested manually on Windows that it now works.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBSyncThread.cpp [hunk 1]
```diff
@@ -75,7 +75,6 @@ Result RocksDBSyncThread::syncWal() {
 }
 
 Result RocksDBSyncThread::sync(rocksdb::DB* db) {
-#ifndef _WIN32
   // if called on Windows, we would get the following error from RocksDB:
   // > Not implemented: SyncWAL() is not supported for this implementation of
   // WAL file
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBSyncThread.cpp:80]: This comment is now also obsolete.

File: arangod/RocksDBEngine/RocksDBTransactionState.cpp [hunk 2]
```diff
@@ -429,7 +418,6 @@ arangodb::Result RocksDBTransactionState::internalCommit() {
   commitCounts();
   cleanupCollTrx.cancel();
 
-#ifndef _WIN32
   // wait for sync if required, for all other platforms but Windows
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBTransactionState.cpp:421]:
```suggestion
  // wait for sync if required
```

**Review by mpoeter — APPROVED** [review]
I have tested this on my Windows machine and it now works as intended. I have also checked the RocksDB code and the `IsSyncThreadSafe()` function which is used to determine whether the old error `SyncWAL() is not supported for this implementation of WAL file` should be returned is now implemented as follows:
```

File: arangod/RocksDBEngine/Methods/RocksDBTrxBaseMethods.cpp [hunk 3]
```diff
@@ -387,7 +376,6 @@ arangodb::Result RocksDBTrxBaseMethods::doCommit() {
   _state->commitCollections(_lastWrittenOperationTick);
   cleanupCollTrx.cancel();
 
-#ifndef _WIN32
   // wait for sync if required, for all other platforms but Windows
```
Comment (jsteemann) [inline arangod/RocksDBEngine/Methods/RocksDBTrxBaseMethods.cpp:379]:
```suggestion
  // wait for sync if required
```

File: arangod/RocksDBEngine/RocksDBSyncThread.cpp (hunk 1 above)
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBSyncThread.cpp:80]:
```suggestion
```

**Pull request merged**